iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/pointer_repository.hpp

```diff
@@ -38,142 +42,71 @@ class PointerRepository
-    static constexpr size_t MIN_ID = 1u;
-    // remark: 0 is a special purpose id and reserved
-    // id 0 is reserved to interpret the offset just as a raw pointer,
-    // i.e. its corresponding base ptr is 0
+    /// @note 0 is a special purpose id and reserved
+    /// id 0 is reserved to interpret the offset just as a raw pointer,
+    /// i.e. its corresponding base ptr is 0
+    static constexpr size_t MIN_ID = 1U;
+    static constexpr size_t MAX_ID = CAPACITY - 1U;
 
   public:
     static constexpr id_t INVALID_ID = std::numeric_limits<id_t>::max();
 
-    PointerRepository()
-        : m_info(CAPACITY)
-    {
-    }
+    /// @brief default constructor
+    PointerRepository() noexcept;
 
-    bool registerPtr(id_t id, ptr_t ptr, uint64_t size)
-    {
-        if (id > MAX_ID)
-        {
-            return false;
-        }
-        if (m_info[id].basePtr == nullptr)
-        {
-            m_info[id].basePtr = ptr;
-            m_info[id].endPtr = reinterpret_cast<ptr_t>(reinterpret_cast<uint64_t>(ptr) + size - 1u);
-            if (id > m_maxRegistered)
-            {
-                m_maxRegistered = id;
-            }
-            return true;
-        }
-        return false;
-    }
-
-    id_t registerPtr(const ptr_t ptr, uint64_t size = 0u)
-    {
-        for (id_t id = 1u; id <= MAX_ID; ++id)
-        {
-            if (m_info[id].basePtr == nullptr)
-            {
-                m_info[id].basePtr = ptr;
-                m_info[id].endPtr = reinterpret_cast<ptr_t>(reinterpret_cast<uint64_t>(ptr) + size - 1u);
-                if (id > m_maxRegistered)
-                {
-                    m_maxRegistered = id;
-                }
-                return id;
-            }
-        }
-
-        return INVALID_ID;
-    }
-
-    bool unregisterPtr(id_t id)
-    {
-        if (id <= MAX_ID && id >= MIN_ID)
-        {
-            if (m_info[id].basePtr != nullptr)
-            {
-                m_info[id].basePtr = nullptr;
+    /// @brief registers the start pointer of the segment in another application with a specific id
+    /// @param[in] id identifies the segment
+    /// @param[in] ptr is the start pointer of the segment
+    /// @param[in] size is the size of the segment
+    /// @return true if the registration was successful, otherwise false
+    bool registerPtr(id_t id, ptr_t ptr, uint64_t size) noexcept;
 
-                // do not search for next lower registered index but we could do it here
-                return true;
-            }
-        }
+    /// @brief registers the start pointer of a segment with a specific size
+    /// @param[in] ptr is the start pointer of the segment
+    /// @param[in] size is the size of the segment
+    /// @return the id that identifies the segment
+    id_t registerPtr(const ptr_t ptr, uint64_t size = 0U) noexcept;
 
-        return false;
-    }
+    /// @brief unregisters the id and also invalidates relative pointers using this id
```

Comment:
It does not really invalidate the relative pointers in use (it cannot, it does not know them). It would be better to have an @attention were we state that relative pointers corresponding to this id become unsafe to use (they can be considered invalid).